akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -367,6 +383,24 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
   }
 
   /**
+   * Java API: The supplied thunk will be run, once, when current cluster member is `Removed`. Used w/ error exit.
+   * If the cluster has already been shutdown the thunk will run on the caller thread immediately.
+   * Typically used together `cluster.leave(cluster.selfAddress)` and then `system.terminate()`.
+   */
+  def registerOnMemberRemovedAfterDown(callback: Runnable): Unit = {
+    registerOnMemberRemoved(callback)
+  }
+
+  /**
+   * Java API: The supplied thunk will be run, once, when current cluster member is `Removed`. Used w/ graceful exit.
+   * If the cluster has already been shutdown the thunk will run on the caller thread immediately.
+   * Typically used together `cluster.leave(cluster.selfAddress)` and then `system.terminate()`.
+   */
+  def registerOnMemberRemovedAfterLeaving(callback: Runnable): Unit = {
+    registerOnMemberRemoved(callback)
```

Comment:
but how is this different from `registerOnMemberRemoved`?
You would have to implement this in the actor and check the `previousStatus`.
I'm not sure how useful these new callbacks are. I think that for more advanced scenarios you should create your own actor instead and listen to the membership events.